### PR TITLE
(Fixes #360) Ignoring AMI version if release label is set and vice-versa

### LINKED ISSUE #360: Unable to create MapReduceCluster with release Label.
Example:
```
val emrCluster = MapReduceCluster()
      .withTaskInstanceCount(32)
      .withReleaseLabel("emr-4.4.0")
```

We receive the following error:
```
[main] ERROR com.krux.hyperion.HyperionAwsClientForPipelineDef - Either AMI Version (or) Release Label should be specified, not both
[main] ERROR com.krux.hyperion.HyperionAwsClientForPipelineDef - Failed to create pipeline
[main] ERROR com.krux.hyperion.HyperionAwsClientForPipelineDef - Deleting the just created pipeline
```

### PR DESCRIPTION
**Pull request by arijitdey**
We were unable to create MapReduceCluster with release Label as it was throwing the following error - Either AMI Version (or) Release Label should be specified, not both

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ trait EmrCluster extends ResourceObject {
 
   def amiVersion = emrClusterFields.amiVersion
   def withAmiVersion(version: HString): Self = updateEmrClusterFields(
-    emrClusterFields.copy(amiVersion = (Option(version)))
+    emrClusterFields.copy(amiVersion = (Option(version)), releaseLabel = None)
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: remove the extra parenthesis in `(Option(version))`

**Comment by realstraw** [comment]
:+1: can you squash?

**Comment by arijitdey** [comment]
@realstraw I already squashed all my commits. I guess merge from master can't be squashed.

**Comment by realstraw** [comment]
@arijitdey ok, please try to branch out from the latest master next time or do rebase.

**Comment by arijitdey** [comment]
@realstraw Will keep in mind next time. Thanks!

**Pull request merged**